<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-0.9.318">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="dcterms.date" content="2022-07-11">

<title>Hands-on introduction to protein modeling - 3&nbsp; Homology Modeling</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<link href="./references.html" rel="next">
<link href="./intro.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link id="quarto-text-highlighting-styles" href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


</head>

<body class="floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
    <div class="container-fluid d-flex justify-content-between">
      <h1 class="quarto-secondary-nav-title"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Homology Modeling</span></h1>
      <button type="button" class="quarto-btn-toggle btn" aria-label="Show secondary navigation">
        <i class="bi bi-chevron-right"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Hands-on introduction to protein modeling</a> 
    </div>
      </div>
      <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Course Disclaimer</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./intro.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Introduction</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./homology.html" class="sidebar-item-text sidebar-link active"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Homology Modeling</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./references.html" class="sidebar-item-text sidebar-link">References</a>
  </div>
</li>
    </ul>
    </div>
    <nav id="TOC" role="doc-toc">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#modeling-protein-structures-mind-the-gap" id="toc-modeling-protein-structures-mind-the-gap" class="nav-link active" data-scroll-target="#modeling-protein-structures-mind-the-gap"> <span class="header-section-number">4</span> Modeling protein structures: Mind the gap</a>
  <ul class="collapse">
  <li><a href="#are-protein-structures-predictable-at-all" id="toc-are-protein-structures-predictable-at-all" class="nav-link" data-scroll-target="#are-protein-structures-predictable-at-all"> <span class="header-section-number">4.1</span> Are protein structures predictable at all?</a></li>
  </ul></li>
  <li><a href="#homology-modeling-conservation-is-the-key" id="toc-homology-modeling-conservation-is-the-key" class="nav-link" data-scroll-target="#homology-modeling-conservation-is-the-key"> <span class="header-section-number">5</span> Homology modeling: Conservation is the key</a></li>
  <li><a href="#homology-modeling-in-four-steps" id="toc-homology-modeling-in-four-steps" class="nav-link" data-scroll-target="#homology-modeling-in-four-steps"> <span class="header-section-number">6</span> Homology modeling in four steps</a>
  <ul class="collapse">
  <li><a href="#step-12.-template-search-and-align." id="toc-step-12.-template-search-and-align." class="nav-link" data-scroll-target="#step-12.-template-search-and-align."> <span class="header-section-number">6.1</span> Step 1+2. Template search and align.</a>
  <ul class="collapse">
  <li><a href="#where-can-we-search" id="toc-where-can-we-search" class="nav-link" data-scroll-target="#where-can-we-search"> <span class="header-section-number">6.1.0.1</span> Where can we search?</a></li>
  <li><a href="#how-can-we-search-accurately-and-fast" id="toc-how-can-we-search-accurately-and-fast" class="nav-link" data-scroll-target="#how-can-we-search-accurately-and-fast"> <span class="header-section-number">6.1.0.2</span> How can we search accurately and fast?</a></li>
  </ul></li>
  <li><a href="#step-3.-model-building." id="toc-step-3.-model-building." class="nav-link" data-scroll-target="#step-3.-model-building."> <span class="header-section-number">6.2</span> Step 3. Model Building.</a></li>
  <li><a href="#step-4.-result-assessing." id="toc-step-4.-result-assessing." class="nav-link" data-scroll-target="#step-4.-result-assessing."> <span class="header-section-number">6.3</span> Step 4. Result assessing.</a></li>
  </ul></li>
  <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references"> <span class="header-section-number">7</span> References</a></li>
  </ul>
</nav>
</nav>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title d-none d-lg-block"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Homology Modeling</span></h1>
</div>





<div class="quarto-title-meta">

    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">July 11, 2022</p>
    </div>
  </div>
    
  </div>
  

</header>

<section id="modeling-protein-structures-mind-the-gap" class="level1" data-number="4">
<h1 data-number="4"><span class="header-section-number">4</span> Modeling protein structures: Mind the gap</h1>
<p>The number of protein sequences in databases growth exponentially during the last decades, particularly after the revolution of high throughput sequencing methods.</p>
<div id="fig-anonymous-1" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><a href="https://www.ebi.ac.uk/uniprot/TrEMBLstats"><img src="pics/trembl_stats.png" title="Number of entries in UniProt/TrEMBL" class="img-fluid figure-img"></a></p>
<p></p><figcaption aria-hidden="true" class="figure-caption">Number of entries in UniProt/TrEMBL (release 2022_2).</figcaption><p></p>
</figure>
</div>
<p>However, experimental determination of 3D protein structures is often difficult, time-consuming and subjected to experimental error, data interpretation and modeling. Thus, despite substantial efforts that started at the beginning of the 21st century to implement high-throughput structural biology methods (see for instance <span class="citation" data-cites="manjasetty2008">Manjasetty et al. (<a href="#ref-manjasetty2008" role="doc-biblioref">2008</a>)</span>), the availability of protein structures is more than 1,000 times less than the number of sequences (231,354,261 sequences for Uniprot and 192,489 structures in RCSB Protein Databank in 2022). This difference is called the <em>protein structure gap</em> and it is constantly widening (<span class="citation" data-cites="muhammed2019">Muhammed and Aki-Yalcin (<a href="#ref-muhammed2019" role="doc-biblioref">2019</a>)</span>).</p>
<div id="fig-anonymous-2" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><a href="https://www.rcsb.org/stats/growth/growth-released-structures"><img src="pics/RCSB_stats.png" title="Number of macromolecular structures in RCSB PDB database." class="img-fluid figure-img"></a></p>
<p></p><figcaption aria-hidden="true" class="figure-caption">Number of macromolecular structures in RCSB PDB database (accessed 6th July 2022).</figcaption><p></p>
</figure>
</div>
<p>Thus, an accurate prediction of the 3D structure of any given protein may be needed to make up for the lack of experimental data.</p>
<section id="are-protein-structures-predictable-at-all" class="level2" data-number="4.1">
<h2 data-number="4.1" class="anchored" data-anchor-id="are-protein-structures-predictable-at-all"><span class="header-section-number">4.1</span> Are protein structures predictable at all?</h2>
<p>We know that amino acids properties determine the phi and psi angles that eventually shape the higher structural levels. Protein folding might be more complex though, as it should be coupled to protein synthesis.</p>
<p>In 1968, Cyrus Levinthal (1922–1990) published the so-called <em>Levinthal’s paradox</em>, stating that proteins fold in nano/milliseconds, but even for small peptides it will take a huge time to test the astronomical number of possible conformations. Say a 100 aa small protein; it will have 99 peptidic bonds and 198 different phi and psi angles. Assuming only 3 alternative conformations for each bond, it will yield 3<sup>198</sup> possible conformations. If we design a highly efficient algorithm that test 1 conformation per nanosecond:</p>
<p>&nbsp;2.95 x 1085 secs = 9x10<sup>67</sup> billions years</p>
<p>Considering that the age of the universe is 13.8 billions years, predicting protein structures does not seem an easy task.</p>
<p>In this context, 50 years ago, a very simple experiment led some light in the protein folding mechanism. <a href="https://en.wikipedia.org/wiki/Christian_B._Anfinsen">Cristian Anfisen</a> was able to completely denature (unfold) a ribonuclease by the addition of reducing agents and urea under heat treatment and subsequently switch to normal conditions that allow the protein to re-fold fully functional. This experiment indicates that amino acid sequence dictates the final structure. Notwithstanding some relevant exceptions, this has been largely confirmed.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figure class="figure">
<img src="pics/anfinsen.jpg" title="Anfinsen" class="img-fluid figure-img">
</figure>
<p></p><figcaption aria-hidden="true" class="figure-caption">The Anfinsen Dogma: Amino acid sequence dictated the final structure. From <span class="citation" data-cites="anfinsen1973">Anfinsen (<a href="#ref-anfinsen1973" role="doc-biblioref">1973</a>)</span> .</figcaption><p></p>
</figure>
</div>
<p>One can imagine that <em>in vivo</em> native structure of proteins look alike the lowest free energy conformation, i.e., the global energy minimum. That is the basis of the funnel model of protein folding, that assumes that the number of possible conformations is reduced when a local energy minimum are achieved, constituting a path for the folding process.</p>
<div id="fig-anonymous-3" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><a href="https://www.sciencedirect.com/science/article/pii/S0968000400017072?casa_token=BR_maX_GNYYAAAAA:IZrQqI9jIbiv-ZeA6aOMHQxsVcp-wgy0XPFO3DRcFiAi7TSrX-3cc7Jb6dhTHdXSQseEhF3l#BIB49"><img src="pics/funnel_1.jpeg" title="The funnel of protein folding" class="img-fluid figure-img"></a></p>
<p></p><figcaption aria-hidden="true" class="figure-caption">Schematic diagram of a protein folding energy landscape according to the funnel model. Denatured molecules at the top of the funnel might fold to the native state by a myriad of different routes, some of which involve transient intermediates (local energy minima) whereas others involve significant kinetic traps (misfolded states). From ; .</figcaption><p></p>
</figure>
</div>
<p>In conclusion, the prediction of protein structures is possible as protein folding relies only on the protein sequence, but it will require virtually infinite computational resources.</p>
</section>
</section>
<section id="homology-modeling-conservation-is-the-key" class="level1" data-number="5">
<h1 data-number="5"><span class="header-section-number">5</span> Homology modeling: Conservation is the key</h1>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figure class="figure">
<img src="pics/modeling.png" title="Protein modeling" class="img-fluid figure-img">
</figure>
<p></p><figcaption aria-hidden="true" class="figure-caption">Protein modeling in a nutshell</figcaption><p></p>
</figure>
</div>
<p>How we do it? Traditionally and conceptually, prediction of protein structures can be addressed from two different perspectives: comparative modeling and <em>ab initio</em> prediction. In the comparative modeling approach, one can predict 3D structure for protein sequences only if their homologs are found in the database of proteins with known structures. Ab initio prediction has no such limitation as it based in the physicochemical properties of the protein molecule, but its accuracy is not high.</p>
<div id="fig-anonymous-4" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><a href="http://isw3.naist.jp/IS/Bio-Info-Unit/gogroup/study/study-en.html"><img src="pics/predway.jpeg" title="Two different approaches for structure prediction." class="img-fluid figure-img"></a></p>
<p></p><figcaption aria-hidden="true" class="figure-caption">Two different approaches for structure prediction. From &lt;http://isw3.naist.jp/IS/Bio-Info-Unit/gogroup/study/study-en.html&gt;</figcaption><p></p>
</figure>
</div>
<p>The accuracy of homology modeling will be limited by the availability of similar structures, <em>ab initio</em> predictions are limited by the mathematical models and computational resources, being often useful only with small peptides.</p>
<p>Homology modeling approaches assume that a similar protein sequence involves a similar 3D structure and function. At the same time, although the number of published sequences and structures continually increased, the number of unique folds remains almost steady since 2008.</p>
<div id="fig-anonymous-5" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><a href="https://www.bonvinlab.org/education/molmod_online/modelling/"><img src="pics/rcsb-statistics.png" title="Protein structures and unique folds" class="img-fluid figure-img"></a></p>
<p></p><figcaption aria-hidden="true" class="figure-caption">Growth of the protein structure database since its inception in 1974. From [https://www.bonvinlab.org/education/molmod_online/modelling/](https://www.bonvinlab.org/education/molmod_online/modelling/https://www.bonvinlab.org/education/molmod_online/modelling/)</figcaption><p></p>
</figure>
</div>
<p>That implies that the space of protein sequences is much larger than the space of structures. This has been exploited by some structural databases, like <a href="https://scop.mrc-lmb.cam.ac.uk/">Scop2</a> or <a href="https://www.cathdb.info/">CATH</a>, that use hierarchical classifications of structures into a very few categories of different structures.</p>
<div id="fig-anonymous-6" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><a href="https://www.cathdb.info/wiki/doku/?id=release_notes"><img src="pics/cath.png" title="CATH" class="img-fluid figure-img"></a></p>
<p></p><figcaption aria-hidden="true" class="figure-caption">CATH main categories exemplified the limited structures space, with 86k vs 500k domains in 2006 and 2022, respectively, but 40 and 41 architectures in the same years.</figcaption><p></p>
</figure>
</div>
<p>In conclusion, protein structures are more conserved than sequences, which allows for model construction by comparison of related proteins. Biological sequences evolve through mutation and selection and selective pressure is different for each residue position in a protein according to its structural and functional relevance. Sequence alignments attempt to tell us the evolutionary story of proteins.</p>
</section>
<section id="homology-modeling-in-four-steps" class="level1" data-number="6">
<h1 data-number="6"><span class="header-section-number">6</span> Homology modeling in four steps</h1>
<div id="fig-anonymous-7" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><a href="https://swissmodel.expasy.org/static/course/files/PartII_homology_modelling.pdf"><img src="pics/homology_simple_workflow.png" title="Homology Modeling workflow" class="img-fluid figure-img"></a></p>
<p></p><figcaption aria-hidden="true" class="figure-caption">Workflow of template-based protein modeling. From [Expasy Protein Structure, Comparative Protein Modelling, and Visualisation](https://swissmodel.expasy.org/course)</figcaption><p></p>
</figure>
</div>
<p>A basic workflow of homology modeling requires three steps: (1) identification of the best-suitable template, (2) alignment of the query sequence and the template, and (3) construction of the model. These steps can be addressed with diverse methodological alternatives and can provide different outputs that must be assessed (step 4) in order to decide the best solution for each step.</p>
<p>In this course, we are focusing on end-to-end modeling with <a href="https://swissmodel.expasy.org/">SWISS-MODEL</a> (<span class="citation" data-cites="waterhouse2018">Waterhouse et al. (<a href="#ref-waterhouse2018" role="doc-biblioref">2018</a>)</span>), a fully automated modeling server that allows you the construction of models without the requirement of a strong bioinformatics background or coding skills. <span style="color:red">Rather than provide just a recipe to make a model in two clics, the goal is understand the process behind, while doing exercises with selected examples</span></p>
<p>As an examples, we are modeling a human DNA repair protein, <strong>NEIL2</strong> and a <strong>viral DNA polymerase</strong> (HAdV-2 pol).</p>
<p>SWISS-MODEL has different supported inputs. Usually, you provide a single sequence query for template searching, but you can bypass this step and provide directly the template you want to use and even a template and a custom alignment. We will start with the first alternative.</p>
<section id="step-12.-template-search-and-align." class="level2" data-number="6.1">
<h2 data-number="6.1" class="anchored" data-anchor-id="step-12.-template-search-and-align."><span class="header-section-number">6.1</span> Step 1+2. Template search and align.</h2>
<section id="where-can-we-search" class="level4" data-number="6.1.0.1">
<h4 data-number="6.1.0.1" class="anchored" data-anchor-id="where-can-we-search"><span class="header-section-number">6.1.0.1</span> Where can we search?</h4>
<p>Template searching consists in finding a protein with known structure that has a sequence similar to our protein. As we mentioned above, the RCSB Protein Data Bank is the largest database of protein structures, so we can search templates by comparing the sequence of our protein with the sequence of all the proteins in PDB. However, PDB was constructed to contain all the macromolecular structures, not for searching templates for modeling. As other end-to-end softwares, SWISS-MODEL has its own curated database, called <strong>SMLT</strong> (SWISS-MODEL template library). This is based on PDB, updated weekly and also annotated and indexed to boost searches. As of July 6th 2022, SMLT contains 133,049 unique protein sequences that map to 332,864 biological units, i.e., quaternary protein structures.</p>
</section>
<section id="how-can-we-search-accurately-and-fast" class="level4" data-number="6.1.0.2">
<h4 data-number="6.1.0.2" class="anchored" data-anchor-id="how-can-we-search-accurately-and-fast"><span class="header-section-number">6.1.0.2</span> How can we search accurately and fast?</h4>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figure class="figure">
<img src="pics/template_query.png" title="Query-template alignment is the base for a new model" class="img-fluid figure-img">
</figure>
<p></p><figcaption aria-hidden="true" class="figure-caption">Query-template alignment is the base for homology modeling (<span class="citation" data-cites="kelley2009">Kelley (<a href="#ref-kelley2009" role="doc-biblioref">2009a</a>)</span>)</figcaption><p></p>
</figure>
</div>
<p>Finding templates require compare sequences, thus an accurate and powerful alignment method is essential. Compare one protein sequence with a whole database is time-consuming, as you will compare with totally unrelated proteins, which is a loss of resources. Two basic improvements increased the template search capacity, (1) the introduction of secondary structure (SS) by comparing SS predictions of the query protein and the secondary structures of the protein database and, (2) the use of profiles to make easier the comparison. Profiles are a mathematical way to summarize a multiple sequence alignment in which the frequency of each amino acid at each position is quantified. This allows the identification of highly conserved positions that not only define the protein function but also the fold. For instance, glycines at the end of each beta-strand or a pattern of polar residues that favor alpha-helices. A previous comparison of the query sequence with a database of sequence will allow us to include evolutionary information about it. Moreover, generation of profiles also facilitates clustering the search database, reducing search time. The implementation of this capacities led to the implementation of <strong>fold recognition</strong> in homology modeling.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figure class="figure">
<img src="pics/profiles.jpg" title="From sequence vs. sequence search to profile-profile comparison" class="img-fluid figure-img">
</figure>
<p></p><figcaption aria-hidden="true" class="figure-caption">From sequence vs.&nbsp;sequence search to profile-profile comparison (<span class="citation" data-cites="kelley2009a">Kelley (<a href="#ref-kelley2009a" role="doc-biblioref">2009b</a>)</span>)</figcaption><p></p>
</figure>
</div>
<p>Template searches in SWISS-MODEL are carried out with HHblits (<span class="citation" data-cites="remmert2011">Remmert et al. (<a href="#ref-remmert2011" role="doc-biblioref">2011</a>)</span>), a specific profile-profile method. We could search for templates also with Blast or other profile-profile methods, like <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi?PAGE_TYPE=BlastSearch&amp;PROGRAM=blastp&amp;BLAST_PROGRAMS=psiBlast">Psi-BLAST</a>, <a href="https://toolkit.tuebingen.mpg.de/tools/hhpred">HHPred</a> or <a href="https://www.ebi.ac.uk/Tools/hmmer/search/jackhmmer">JackHMMER</a>. Then templates are ranked by two different parameters, <strong>GMQE</strong> (global model quality estimate) and <strong>QSQE</strong> (quaternary structure quality estimate). Briefly, GMQE uses probability functions to assess several properties of the target-template alignment (sequence identity, sequence similarity, HHblits score, agreement between predicted secondary structure of target and template, agreement between predicted solvent accessibility between target and template; all normalized by alignment length) to predict the expected quality of the resulting model. QSQE assess the oligomeric state probability of the model.</p>
</section>
</section>
<section id="step-3.-model-building." class="level2" data-number="6.2">
<h2 data-number="6.2" class="anchored" data-anchor-id="step-3.-model-building."><span class="header-section-number">6.2</span> Step 3. Model Building.</h2>
<p>adsf</p>
</section>
<section id="step-4.-result-assessing." class="level2" data-number="6.3">
<h2 data-number="6.3" class="anchored" data-anchor-id="step-4.-result-assessing."><span class="header-section-number">6.3</span> Step 4. Result assessing.</h2>
<div id="fig-anonymous-8" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1007449#sec002"><img src="pics/ten.png" title="Homology modeling" class="img-fluid figure-img"></a></p>
<p></p><figcaption aria-hidden="true" class="figure-caption">Full homology-modeling workflow (from ).</figcaption><p></p>
</figure>
</div>
</section>
</section>
<section id="references" class="level1" data-number="7">
<h1 data-number="7"><span class="header-section-number">7</span> References</h1>


<div id="refs" class="references csl-bib-body hanging-indent" role="doc-bibliography">
<div id="ref-anfinsen1973" class="csl-entry" role="doc-biblioentry">
Anfinsen, C. B. 1973. <span>“Principles that govern the folding of protein chains.”</span> <em>Science (New York, N.Y.)</em> 181 (4096): 223–30. <a href="https://doi.org/10.1126/science.181.4096.223">https://doi.org/10.1126/science.181.4096.223</a>.
</div>
<div id="ref-kelley2009" class="csl-entry" role="doc-biblioentry">
Kelley, Lawrence A. 2009a. <span>“Fold Recognition.”</span> In, edited by Daniel John Rigden, 27–55. Dordrecht: Springer Netherlands. <a href="https://doi.org/10.1007/978-1-4020-9058-5_2">https://doi.org/10.1007/978-1-4020-9058-5_2</a>.
</div>
<div id="ref-kelley2009a" class="csl-entry" role="doc-biblioentry">
———. 2009b. <span>“Fold Recognition.”</span> In, edited by Daniel John Rigden, 27–55. Dordrecht: Springer Netherlands. <a href="https://doi.org/10.1007/978-1-4020-9058-5_2">https://doi.org/10.1007/978-1-4020-9058-5_2</a>.
</div>
<div id="ref-manjasetty2008" class="csl-entry" role="doc-biblioentry">
Manjasetty, Babu A., Andrew P. Turnbull, Santosh Panjikar, Konrad Büssow, and Mark R. Chance. 2008. <span>“Automated Technologies and Novel Techniques to Accelerate Protein Crystallography for Structural Genomics.”</span> <em>PROTEOMICS</em> 8 (4): 612–25. <a href="https://doi.org/10.1002/pmic.200700687">https://doi.org/10.1002/pmic.200700687</a>.
</div>
<div id="ref-muhammed2019" class="csl-entry" role="doc-biblioentry">
Muhammed, Muhammed Tilahun, and Esin Aki-Yalcin. 2019. <span>“Homology modeling in drug discovery: Overview, current applications, and future perspectives.”</span> <em>Chemical Biology &amp; Drug Design</em> 93 (1): 12–20. <a href="https://doi.org/10.1111/cbdd.13388">https://doi.org/10.1111/cbdd.13388</a>.
</div>
<div id="ref-remmert2011" class="csl-entry" role="doc-biblioentry">
Remmert, Michael, Andreas Biegert, Andreas Hauser, and Johannes Söding. 2011. <span>“HHblits: lightning-fast iterative protein sequence searching by HMM-HMM alignment.”</span> <em>Nature Methods</em> 9 (2): 173–75. <a href="https://doi.org/10.1038/nmeth.1818">https://doi.org/10.1038/nmeth.1818</a>.
</div>
<div id="ref-waterhouse2018" class="csl-entry" role="doc-biblioentry">
Waterhouse, Andrew, Martino Bertoni, Stefan Bienert, Gabriel Studer, Gerardo Tauriello, Rafal Gumienny, Florian T Heer, et al. 2018. <span>“SWISS-MODEL: Homology Modelling of Protein Structures and Complexes.”</span> <em>Nucleic Acids Research</em> 46 (W1): W296–303. <a href="https://doi.org/10.1093/nar/gky427">https://doi.org/10.1093/nar/gky427</a>.
</div>
</div>
</section>

</main> <!-- /main -->
<script type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      let href = ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
    var links = window.document.querySelectorAll('a:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
          // target, if specified
          link.setAttribute("target", "_blank");
          // default icon
          link.classList.add("external");
      }
    }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./intro.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Introduction</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./references.html" class="pagination-link">
        <span class="nav-page-text">References</span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->



</body></html>