<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-0.9.318">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Modesto Redrejo Rodríguez">
<meta name="dcterms.date" content="2022-08-25">

<title>Structural Bioinformatics 2022-23 - Introduction</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link id="quarto-text-highlighting-styles" href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


</head>

<body class="docked">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
    <div class="container-fluid d-flex justify-content-between">
      <h1 class="quarto-secondary-nav-title">Introduction</h1>
      <button type="button" class="quarto-btn-toggle btn" aria-label="Show secondary navigation">
        <i class="bi bi-chevron-right"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Structural Bioinformatics 2022-23</a> 
        <div class="sidebar-tools-main tools-wide">
    <a href="https://twitter.com/mredrejo" title="" class="sidebar-tool px-1"><i class="bi bi-twitter"></i></a>
    <a href="mailto:modesto.redrejo@uam.es" title="" class="sidebar-tool px-1"><i class="bi bi-envelope-paper"></i></a>
    <a href="https://github.com/strBio/strbio2022_23" title="" class="sidebar-tool px-1"><i class="bi bi-github"></i></a>
</div>
    </div>
      </div>
      <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item sidebar-item-section">
    <div class="sidebar-item-container"> 
        <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">Basics</a>
      <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
        <i class="bi bi-chevron-right ms-2"></i>
      </a>
    </div>
    <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">About StrBio 22-23</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./intro.html" class="sidebar-item-text sidebar-link active">Introduction</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./ddbb.html" class="sidebar-item-text sidebar-link">Protein Databases</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./pdb.html" class="sidebar-item-text sidebar-link">Working with protein structures</a>
  </div>
</li>
    </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
    <div class="sidebar-item-container"> 
        <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">Modeling</a>
      <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">
        <i class="bi bi-chevron-right ms-2"></i>
      </a>
    </div>
    <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./homology.html" class="sidebar-item-text sidebar-link">Homology Modeling</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./ai.html" class="sidebar-item-text sidebar-link">State-of-the-art protein modeling with Deep Learning-based methods</a>
  </div>
</li>
    </ul>
  </li>
    </ul>
    </div>
</nav>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#goals-and-warnings" id="toc-goals-and-warnings" class="nav-link active" data-scroll-target="#goals-and-warnings">Goals and Warnings</a></li>
  <li><a href="#warning-for-current-and-future-structural-biologists" id="toc-warning-for-current-and-future-structural-biologists" class="nav-link" data-scroll-target="#warning-for-current-and-future-structural-biologists">Warning for current and future structural biologists</a></li>
  <li><a href="#before-going-forward-protein-structure-101" id="toc-before-going-forward-protein-structure-101" class="nav-link" data-scroll-target="#before-going-forward-protein-structure-101">Before going forward: Protein Structure 101</a>
  <ul class="collapse">
  <li><a href="#ramachandran-plot" id="toc-ramachandran-plot" class="nav-link" data-scroll-target="#ramachandran-plot">Ramachandran Plot</a></li>
  <li><a href="#protein-folds-domains-and-motifs" id="toc-protein-folds-domains-and-motifs" class="nav-link" data-scroll-target="#protein-folds-domains-and-motifs">Protein folds, domains and motifs</a></li>
  <li><a href="#hands-on-playing-with-secondary-structures" id="toc-hands-on-playing-with-secondary-structures" class="nav-link" data-scroll-target="#hands-on-playing-with-secondary-structures"><span style="color:green"><strong>Hands on: Playing with secondary structures</strong></span></a></li>
  </ul></li>
  <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references">References</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title d-none d-lg-block">Introduction</h1>
</div>





<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Modesto Redrejo Rodríguez </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">August 25, 2022</p>
    </div>
  </div>
    
  </div>
  

</header>

<section id="goals-and-warnings" class="level2" data-link="Preface">
<h2 data-link="Preface" class="anchored" data-anchor-id="goals-and-warnings">Goals and Warnings</h2>
<p><a href="https://en.wikipedia.org/wiki/Structural_bioinformatics" title="Wikipedia">Structural Bioinformatics</a> (SB) s a broad discipline that covers structural and computational biology, from visualization and analysis of the structure of biomacromolecules to protein modeling and molecular docking. The great promise of SB is predicated on the belief that a high-resolution structural information about biological systems will allow us to precisely reason about the function of these systems and the effects of modifications and perturbations.</p>
<p>The goals of SB require at least four different research lines (see Chapter 1 in <span class="citation" data-cites="structura"><span>“Structural Bioinformatics, 2nd Edition | Wiley”</span> (<a href="#ref-structura" role="doc-biblioref">n.d.a</a>)</span>). :</p>
<ol type="1">
<li><p><em>Visualization</em> of complex structures with several sources of information: sequence, structural data, electrostatic fields, location of functional sites, and areas of variability.</p></li>
<li><p><em>Classification</em> of the structures, making if necessary to cluster similar structures together in a hierarchical classification allow us to identify common origins and diversification paths. Similar to other fields of biology classification is tedious but required to understand the structural space.</p></li>
<li><p><em>Prediction</em> of structures remains an area of keen interest and a field of research itself. As we will see below, the number of different sequences is much higher than the availability of structures, which make prediction an essential and useful tool.</p></li>
<li><p><em>Simulation.</em> Experimentally obtained structures are primarily static structural models (see warning below). However, the properties of these molecules are often the results of their dynamic motions. The definition of energy functions that govern the folding of proteins and their subsequent stable dynamics can be analyzed by molecular dynamics simulations, although computation capacities may be limiting to reach a biologically relevant timescales.</p></li>
</ol>
<p>Powered by large amount of data and great technical advances, the field has experienced a great revolution in the last decade. The increase of experimental capacities to analyze the structure of proteins and other biological molecules and structures (see <span class="citation" data-cites="callaway2020">Callaway (<a href="#ref-callaway2020" role="doc-biblioref">2020</a>)</span>) and the development of Artificial Intelligence (AI)-assisted structure prediction boosted the capacity of life-science researchers to address a wide variety of questions regarding proteins diversity, evolution and function. The implications of this revolution in biology, biotechnology, and biomedicine are still unforeseen.</p>
</section>
<section id="warning-for-current-and-future-structural-biologists" class="level2">
<h2 class="anchored" data-anchor-id="warning-for-current-and-future-structural-biologists">Warning for current and future structural biologists</h2>
<div id="fig-magritte" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<figure class="figure">
<img src="pics/magritte.png" title="Ceci n'est pas une proteine" class="figure img-fluid figure-img">
</figure>
<p></p><figcaption aria-hidden="true" class="figure-caption">Figure 1: Ceci n’est pas une proteine. Source: https://swissmodel.expasy.org/static/course/files/PartIII_quality_assesment.pdf</figcaption><p></p>
</figure>
</div>
<p>The surrealist Belgian painter René Magritte created a collection of surrealistic paintings entitled <a href="https://en.wikipedia.org/wiki/The_Treachery_of_Images" title="Magritte"><strong><em>La trahison des images</em></strong> (1928–1929)</a>. The most renowned of those paintings show a smoking pipe and the following caption underneath: <em>“Ceci n’est pas une pipe”</em> (This is not a pipe). Yes, indeed! It is actually the painting of a pipe.</p>
<p>Similarly, a picture of a protein, or a PDB file with the coordinates of a protein structure, is not a protein. It is a representation of ONE structure. Even experimentally determined structures have two main limitations that we should always keep in mind: (1) they are a fixed structure whereas proteins <em>in vivo</em> are flexible and dynamic and (2) they are subjected to experimental error and they often contain regions of low reliability. Moreover, even experimentally obtained macromolecular structures are to some degree models, with a variable ratio between experimental data and computational prediction to match the experimental data (X-ray diffraction, cryo-EM density maps, NMR, SAXS, FRET…) with previously known structures or prediction models. That does not mean that protein structures are useless, they can be very useful, but we must be aware of the limitations as well as the applications.</p>
</section>
<section id="before-going-forward-protein-structure-101" class="level1">
<h1>Before going forward: Protein Structure 101</h1>
<p>Although you can make some protein modeling without being an expert in structural biology, a basic understanding of protein structure is strongly advisable. Over the years, I noticed that graduate students in biology, biomedicine, and related fields have a very different background on protein structure. If you want to review and update your background on protein structure, I recommend you reading Chapter 2 of <span class="citation" data-cites="structur"><span>“Structural Bioinformatics, 2nd Edition | Wiley”</span> (<a href="#ref-structur" role="doc-biblioref">n.d.b</a>)</span>, the great recent review by <span class="citation" data-cites="stollar2020">Stollar and Smith (<a href="#ref-stollar2020" role="doc-biblioref">2020</a>)</span> and the <a href="https://en.wikipedia.org/wiki/Protein_structure">Wikipedia</a> and <a href="https://proteopedia.org/wiki/index.php/Introduction_to_protein_structure">Proteopedia</a> articles on protein structures, which constituted my main source for this brief section.</p>
<div id="fig-str" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><a href="https://en.wikipedia.org/wiki/Protein_structure"><img src="pics/Protein_structure.png" title="Protein Structure" class="img-fluid figure-img"></a></p>
<p></p><figcaption aria-hidden="true" class="figure-caption">Figure 2: Protein structure levels, using human PCNA (PDB 1AXC) as an example.</figcaption><p></p>
</figure>
</div>
<p>Proteins are key components of life, playing key roles in almost any possible vital function, either as structural, or scaffolding elements or as active enzymes that catalyze metabolic reactions. Proteins are built as polymers of amino acids and the sequence of amino acids of a particular protein can be also called the <strong>primary structure</strong> of the protein. Amino acid chains can spontaneously fold up into three-dimensional structures, mostly stabilized by hydrogen bonds between amino acids. The amino acid sequence determines different layers of 3D structure. Each of the 20 natural amino acids has different physicochemical properties that affect its preferred conformation. Thus, the first level of folding is called <strong>secondary structure</strong>, forming common patterns as we will see in a moment.</p>
<div id="fig-aa" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><a href="https://www.reddit.com/r/chemistry/comments/acyald/venn_diagram_showing_the_properties_of_the_20/"><img src="pics/aa.png" title="Amino acids clasification by type" class="img-fluid figure-img"></a></p>
<p></p><figcaption aria-hidden="true" class="figure-caption">Figure 3: Amino acids clasification by type</figcaption><p></p>
</figure>
</div>
<p>These stretches of secondary structure patterns can fold in 3D due to interactions between the side chains of amino acids. This is called protein <strong>tertiary structure</strong>. Finally, two or more individual peptide chains can form multisubunit proteins that have the so-called <strong>quaternary structure</strong>.</p>
<p>It should be noted that the peptide bond itself cannot rotate as it has a double bond-like character. Therefore, rotation can only occur about the bond between the Cα and the C = O group, (the phi (φ) angle) and the Cα and the NH group, (the psi (ψ) angle). In fact, the polypeptide backbone chain is composed of a repeating series of two rotatable bonds followed by one non-rotatable (peptide) bond. However, not all 360º of the psi and phi angles are possible as neighboring sidechains can clash due to steric hindrance. For certain angles and amino acid combinations, the atoms cannot be in the same physical place and this partly explains why some amino acids have a higher propensity (likelihood) to form different types of secondary structures.</p>
<div id="fig-bond" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><a href="https://portlandpress.com/essaysbiochem/article/64/4/649/226515/Uncovering-protein-structure"><img src="pics/peptide_bond.png" title="Peptide bond" class="img-fluid figure-img"></a></p>
<p></p><figcaption aria-hidden="true" class="figure-caption">Figure 4: Scheme of a generic polypeptide chain. Residue side chains are denoted as R. Coloured rectangles indicate sets of six atoms that are coplanar due to the double-bond character of the peptide bond. Arrows indicate the bonds that are free to rotate with the angle of rotation about the N–Cα known as phi and about the Cα–C known as psi. Note that only peptide backbone bonds are labeled and in most cases the R group bond is free to rotate.</figcaption><p></p>
</figure>
</div>
<p>Within these restraints, the two principal local conformations that avoid steric hindrance and maximize backbone–backbone hydrogen bonding are the <strong>α-helix</strong> and the <strong>β-sheet</strong> secondary structures. The α-helix was proposed initially as left-handed by Linus Pauling in 1951, but the crystal structure of myoglobin in 1958 showed that, although both can be found, the right-handed form is the common one. In the common right-handed helices, the backbone NH group hydrogen bonds to the backbone C = O group of the amino acid located four residues earlier along the protein sequence. This results in a polypeptide chain that twists in a regular coil shape with the R-groups pointing outwards away from the peptide backbone. It takes approximately 3.6 residues to complete a full turn of a helix.</p>
<div class="quarto-layout-panel">
<div class="quarto-layout-row quarto-layout-valign-top">
<div id="fig-alpha" class="quarto-figure quarto-figure-center anchored" style="flex-basis: 50.0%;justify-content: center;">
<figure class="figure">
<p><a href="https://en.wikipedia.org/wiki/Alpha_helix"><img src="pics/alpha.gif" title="Alpha helix" class="img-fluid figure-img"></a></p>
<p></p><figcaption aria-hidden="true" class="figure-caption">Figure 5: Alpha helix</figcaption><p></p>
</figure>
</div>
<div id="fig-beta" class="quarto-figure quarto-figure-center anchored" style="flex-basis: 50.0%;justify-content: center;">
<figure class="figure">
<p><a href="https://en.wikipedia.org/wiki/Beta_sheet"><img src="pics/Animated_Beta_sheet.gif" title="Beta sheet" class="img-fluid figure-img"></a></p>
<p></p><figcaption aria-hidden="true" class="figure-caption">Figure 6: Beta sheet</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>Different amino-acid sequences have different propensities for forming α-helical structures. <a href="https://en.wikipedia.org/wiki/Methionine" title="Methionine">Methionine</a>, <a href="https://en.wikipedia.org/wiki/Alanine" title="Alanine">alanine</a>, <a href="https://en.wikipedia.org/wiki/Leucine" title="Leucine">leucine</a>, <a href="https://en.wikipedia.org/wiki/Glutamate" title="Glutamate">glutamate</a>, and <a href="https://en.wikipedia.org/wiki/Lysine" title="Lysine">lysine</a> have especially high helix-forming propensities, whereas <a href="https://en.wikipedia.org/wiki/Proline" title="Proline">proline</a> and <a href="https://en.wikipedia.org/wiki/Glycine" title="Glycine">glycine</a> have poor helix-forming propensities. <a href="https://en.wikipedia.org/wiki/Proline" title="Proline">Proline</a> either breaks or kinks a helix, both because it cannot donate an amide <a href="https://en.wikipedia.org/wiki/Hydrogen_bond" title="Hydrogen bond">hydrogen bond</a> (having no amide hydrogen), and also because its bulky sidechain interferes sterically with the backbone of the preceding turn. However, proline is often seen as the <em>first</em> residue of a helix, it is presumed due to its structural rigidity. At the other extreme, <a href="https://en.wikipedia.org/wiki/Glycine" title="Glycine">glycine</a> also tends to disrupt helices because its high conformational flexibility makes it entropically expensive to adopt the relatively constrained α-helical structure.</p>
<p><strong>β-Sheets</strong> are composed of two or more extended polypeptide chains called β-strands that run alongside each other. They can be arranged in either a parallel or antiparallel manner. The residues arrange themselves in a regular zigzag manner with the adjacent peptide bonds pointing in opposite directions. In this arrangement, the NH group and the C = O group of each amino acid are hydrogen-bonded to the C = O group and NH group respectively on the adjacent strands. Chains can run in opposite directions, forming an antiparallel β-sheet, or in the same direction, forming a parallel β-sheet. Sidechains from each of the residues point away from the sheets and alternate in opposite directions between residues. It is common to see a pattern of alternating hydrophilic and hydrophobic residues in the primary structure, giving the β-sheets hydrophilic and hydrophobic faces.</p>
<p>Large aromatic residues (<a href="https://en.wikipedia.org/wiki/Tyrosine" title="Tyrosine">tyrosine</a>, <a href="https://en.wikipedia.org/wiki/Phenylalanine" title="Phenylalanine">phenylalanine</a>, <a href="https://en.wikipedia.org/wiki/Tryptophan" title="Tryptophan">tryptophan</a>) and β-branched amino acids (<a href="https://en.wikipedia.org/wiki/Threonine" title="Threonine">threonine</a>, <a href="https://en.wikipedia.org/wiki/Valine" title="Valine">valine</a>, <a href="https://en.wikipedia.org/wiki/Isoleucine" title="Isoleucine">isoleucine</a>) are favored to be found in β-strands. As in the case of α-helixes, β-strands are often ended by <a href="https://en.wikipedia.org/wiki/Glycine" title="Glycine">glycines</a>, which are especially common in β-turns (the most common connector between strands), as <a href="https://en.wikipedia.org/wiki/Amino_acid" title="Amino acid">amino acids</a> with positive φ angles.</p>
<section id="ramachandran-plot" class="level3">
<h3 class="anchored" data-anchor-id="ramachandran-plot">Ramachandran Plot</h3>
<p>As you probably already figure out, many combinations of φ and ψ angles are forbidden because of the principle of steric exclusion: two atoms cannot be in the same place ad the same time. This was initially shown by <a href="https://en.wikipedia.org/wiki/G._N._Ramachandran">Gopalasamudram Ramachandran</a>, who also devised a plot to visualize the allowed angle values, so-called Ramachandran plot. This plot can represent the angles of a particular amino acid, of all the amino acids in a protein or many proteins. Analysis of φ and ψ angles in known proteins clearly show that roughly three-quarters of all possible φ, ψ combinations are excluded.</p>
<div id="fig-rama0" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><a href="https://proteopedia.org/wiki/index.php/Ramachandran_Plot"><img src="https://proteopedia.org/wiki/images/9/90/Ramachandran_plot_general_100K.jpg" title="General Ramachandran plot" class="img-fluid figure-img"></a></p>
<p></p><figcaption aria-hidden="true" class="figure-caption">Figure 7: General Ramachandran plot. The density of points reflects how likely is each angle combination, defining the core (red line) and tolerance (orange) regions.</figcaption><p></p>
</figure>
</div>
<p>The core regions in the Rama plot also correspond with common secondary structures, as usually represented in textbooks.</p>
<div id="fig-ram" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<figure class="figure">
<img src="pics/rama2.png" title="Textbook Rama plot" class="figure img-fluid figure-img">
</figure>
<p></p><figcaption aria-hidden="true" class="figure-caption">Figure 8: Definition of secondary structure alternatives by their combination of phi, psi angles.</figcaption><p></p>
</figure>
</div>
<p>Functionally and structurally relevant residues are more likely than others to have torsion angles that plot to the <u>allowed but disfavored</u> regions of a Ramachandran plot. The specific geometry of these functionally relevant residues, while somewhat energetically unfavorable, may be important for the protein’s function, catalytic or otherwise. Such conformations need to be stabilized by the protein using H-bonds, steric packing, or other means, and should very seldom occur for highly solvent-exposed residues.</p>
<div id="fig-rama2" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><a href="https://proteopedia.org/wiki/index.php/Ramachandran_Plot"><img src="https://proteopedia.org/wiki/images/8/85/Ramachandran_Gly_Pro_data_and_contours_T8000_small.jpg" class="img-fluid figure-img"></a></p>
<p></p><figcaption aria-hidden="true" class="figure-caption">Figure 9: Ramachandran plots for glycine (lef) and proline (right) Inner contour encloses 98% and 99.9% of Top structures data, indicating the favored and allowed regions, respectively .</figcaption><p></p>
</figure>
</div>
</section>
<section id="protein-folds-domains-and-motifs" class="level3">
<h3 class="anchored" data-anchor-id="protein-folds-domains-and-motifs">Protein folds, domains and motifs</h3>
<p>The final three dimensional tertiary structure of a protein is commonly referred as its <strong>fold</strong>. Within the overall protein fold, we can recognize distinct <strong>domains</strong> and <strong>motifs.</strong> Domains are compact sections of the protein that represent structurally and (usually) functionally independent regions. That means that a domain maintain its main features, even if separated from the overall protein. On the other hand, motifs are small substructures that are not necessarily independent and consist of only a few secondary structure stretches. Indeed, motifs can be also referred as <em>super-secondary</em> structure.</p>
<p>The diversity of protein folds, domains and motifs, and combination of those, can be used for classification of protein structures hierarchically, as in many other fields of biology. The first classification was proposed in the 70’s and consisted of four groups of folds, as shown in the figure below. All-α proteins are based almost entirely on an α-helical structure, and all β-structure are based on β-sheets. α/β structure is based on am mixture of α-helices and β-sheet, often organized as parallel β-strands connected by α-helices. Finally α+β structures consist of discrete α-helix and β-sheet motifs that are not interwoven (as they are in α/β proteins).</p>
<div id="fig-chlothia" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="pics/clasif.png" class="figure img-fluid figure-img"></p>
<p></p><figcaption aria-hidden="true" class="figure-caption">Figure 10: The four structural protein classes in the classification by Chlothia &amp; Levitt. Modified from <span class="citation" data-cites="structur"><span>“Structural Bioinformatics, 2nd Edition | Wiley”</span> (<a href="#ref-structur" role="doc-biblioref">n.d.b</a>)</span> using 1I2T, 1K76, 1H75 and 1EM7 structures.</figcaption><p></p>
</figure>
</div>
<p>As known fold space has become more and more complex, these types of classifications have been adjusted and extended such that a complete hierarchy is created. The most commonly referred approaches to this sort of classification are those used by SCOP and CATH databases.</p>
</section>
<section id="hands-on-playing-with-secondary-structures" class="level3">
<h3 class="anchored" data-anchor-id="hands-on-playing-with-secondary-structures"><span style="color:green"><strong>Hands on: Playing with secondary structures</strong></span></h3>
<p><img src="pics/handson.png" title="Hands-on" class="img-fluid"></p>
<p>There are a few online alternatives to model any peptide sequence and quickly see the effect of amino acid composition in the secondary structure. One of the best-known is Foldit (<a href="http://www.fold.it">www.fold.it</a>, <span class="citation" data-cites="miller2020">Miller et al. (<a href="#ref-miller2020" role="doc-biblioref">2020</a>)</span>), a gaming platform for biochemistry and structural biology teaching. It is a highly recommended alternative for most courses related to protein structure.</p>
<p>In this course we are going to try a more recent proposal, recently twitted by Sergey Ovchinnikov (see <a href="https://twitter.com/sokrypton/status/1535857255647690753" class="uri">https://twitter.com/sokrypton/status/1535857255647690753</a>). It is based on ColabFold (see <a href="https://github.com/sokrypton/ColabFold" class="uri">https://github.com/sokrypton/ColabFold</a> and <span class="citation" data-cites="mirdita2022">Mirdita et al. (<a href="#ref-mirdita2022" role="doc-biblioref">2022</a>)</span>), an Alphafold2 (see <span class="citation" data-cites="jumper2021">Jumper et al. (<a href="#ref-jumper2021" role="doc-biblioref">2021</a>)</span>) free notebook in <a href="https://colab.research.google.com/?hl=en">Google Colab notebook</a>. All you need is a Google account and the following <em>cheatsheet</em>.</p>
<div id="fig-single" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><a href="https://twitter.com/sokrypton/status/1535857255647690753"><img src="pics/cheatsheet.png" title="Protein hallucination" class="img-fluid figure-img"></a></p>
<p></p><figcaption aria-hidden="true" class="figure-caption">Figure 11: Basic protein amino acids stats for protein design with ColabFold Single</figcaption><p></p>
</figure>
</div>
<p>Now go to ColabFold Single: <a href="https://colab.research.google.com/github/sokrypton/af_backprop/blob/beta/examples/AlphaFold_single.ipynb" class="uri">https://colab.research.google.com/github/sokrypton/af_backprop/blob/beta/examples/AlphaFold_single.ipynb</a></p>
<p>Construct some small proteins and compare the output. Note that the first model will take 3-5 min, but the others will be faster. I provide here some interesting examples (IUPAC one-letter amino acid code):</p>
<ol type="1">
<li><p>AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA</p></li>
<li><p>KKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKK</p></li>
<li><p>PVAVEARENGRLAVRVEGAIAVLIRENGRLVVRVEGG</p></li>
<li><p>PELEKHREELGEFLKKETGIAVEIRENGRLEVRVEGYTDVKIEGGTERLKRFLEEL</p></li>
<li><p>ACTWEGNKLTCA</p></li>
</ol>
<p><span style="color:green"><strong>Answer the following questions:</strong></span></p>
<p><strong><br>
<span style="color:green">- Why is a poly-K more stable (dark blue) than a poly-A?<br>
<br>
- Could you predict the structure of a poly-V or a poly-G?<br>
<br>
- What would happen if you introduce a K5W in the structure number 2? and in the 4?<br>
<br>
Now, try to create peptides with:<br>
<br>
- Two helices<br>
- A four-strands beta-sheet.<br>
- Alpha-beta-beta-alpha.</span></strong><span style="color:green"><br>
</span></p>
<hr>
</section>
</section>
<section id="references" class="level1">
<h1>References</h1>



</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" role="doc-bibliography">
<div id="ref-callaway2020" class="csl-entry" role="doc-biblioentry">
Callaway, Ewen. 2020. <span>“Revolutionary Cryo-EM Is Taking over Structural Biology.”</span> <em>Nature</em> 578 (7794): 201–1. <a href="https://doi.org/10.1038/d41586-020-00341-9">https://doi.org/10.1038/d41586-020-00341-9</a>.
</div>
<div id="ref-jumper2021" class="csl-entry" role="doc-biblioentry">
Jumper, John, Richard Evans, Alexander Pritzel, Tim Green, Michael Figurnov, Olaf Ronneberger, Kathryn Tunyasuvunakool, et al. 2021. <span>“Highly Accurate Protein Structure Prediction with AlphaFold.”</span> <em>Nature</em> 596 (7873): 583–89. <a href="https://doi.org/10.1038/s41586-021-03819-2">https://doi.org/10.1038/s41586-021-03819-2</a>.
</div>
<div id="ref-miller2020" class="csl-entry" role="doc-biblioentry">
Miller, Josh Aaron, Firas Khatib, Haley Hammond, Seth Cooper, and Scott Horowitz. 2020. <span>“Introducing Foldit Education Mode.”</span> <em>Nature Structural &amp; Molecular Biology</em> 27 (9): 769–70. <a href="https://doi.org/10.1038/s41594-020-0485-6">https://doi.org/10.1038/s41594-020-0485-6</a>.
</div>
<div id="ref-mirdita2022" class="csl-entry" role="doc-biblioentry">
Mirdita, Milot, Konstantin Schütze, Yoshitaka Moriwaki, Lim Heo, Sergey Ovchinnikov, and Martin Steinegger. 2022. <span>“ColabFold: making protein folding accessible to all.”</span> <em>Nature Methods</em> 19 (6): 679–82. <a href="https://doi.org/10.1038/s41592-022-01488-1">https://doi.org/10.1038/s41592-022-01488-1</a>.
</div>
<div id="ref-stollar2020" class="csl-entry" role="doc-biblioentry">
Stollar, Elliott J, and David P Smith. 2020. <span>“Uncovering Protein Structure.”</span> <em>Essays in Biochemistry</em> 64 (4): 649–80. <a href="https://doi.org/10.1042/EBC20190042">https://doi.org/10.1042/EBC20190042</a>.
</div>
<div id="ref-structur" class="csl-entry" role="doc-biblioentry">
<span>“Structural Bioinformatics, 2nd Edition | Wiley.”</span> n.d.b. <a href="https://www.wiley.com/en-us/Structural+Bioinformatics%2C+2nd+Edition-p-9780470181058">https://www.wiley.com/en-us/Structural+Bioinformatics%2C+2nd+Edition-p-9780470181058</a>.
</div>
<div id="ref-structura" class="csl-entry" role="doc-biblioentry">
———. n.d.a. <a href="https://www.wiley.com/en-us/Structural+Bioinformatics%2C+2nd+Edition-p-9780470181058">https://www.wiley.com/en-us/Structural+Bioinformatics%2C+2nd+Edition-p-9780470181058</a>.
</div>
</div></section></div></main> <!-- /main -->
<script type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      let href = ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
    var links = window.document.querySelectorAll('a:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
          // target, if specified
          link.setAttribute("target", "_blank");
          // default icon
          link.classList.add("external");
      }
    }
});
</script>
</div> <!-- /content -->



</body></html>