<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-0.9.318">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Modesto Redrejo Rodríguez">
<meta name="dcterms.date" content="2022-09-21">

<title>Structural Bioinformatics 2022-23 - Modeling Resources</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>

<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link id="quarto-text-highlighting-styles" href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<meta name="twitter:title" content="Structural Bioinformatics 2022-23 - Modeling Resources">
<meta name="twitter:card" content="summary">
</head>

<body class="docked">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
    <div class="container-fluid d-flex justify-content-between">
      <h1 class="quarto-secondary-nav-title">Modeling Resources</h1>
      <button type="button" class="quarto-btn-toggle btn" aria-label="Show secondary navigation">
        <i class="bi bi-chevron-right"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Structural Bioinformatics 2022-23</a> 
        <div class="sidebar-tools-main tools-wide">
    <a href="https://twitter.com/mredrejo" title="" class="sidebar-tool px-1"><i class="bi bi-twitter"></i></a>
    <a href="mailto:modesto.redrejo@uam.es" title="" class="sidebar-tool px-1"><i class="bi bi-envelope-paper"></i></a>
    <a href="https://github.com/strBio/strbio2022_23" title="" class="sidebar-tool px-1"><i class="bi bi-github"></i></a>
</div>
    </div>
      </div>
      <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">StrBio 2022-23</a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
    <div class="sidebar-item-container"> 
        <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">Basics</a>
      <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
        <i class="bi bi-chevron-right ms-2"></i>
      </a>
    </div>
    <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./intro.html" class="sidebar-item-text sidebar-link">Introduction</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./ddbb.html" class="sidebar-item-text sidebar-link">Protein Databases</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./pdb.html" class="sidebar-item-text sidebar-link">Protein Structures</a>
  </div>
</li>
    </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
    <div class="sidebar-item-container"> 
        <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">Modeling</a>
      <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">
        <i class="bi bi-chevron-right ms-2"></i>
      </a>
    </div>
    <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./homology.html" class="sidebar-item-text sidebar-link">Homology Modeling</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./advanced.html" class="sidebar-item-text sidebar-link">Advanced homology modeling</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./ai.html" class="sidebar-item-text sidebar-link">State-of-the-art protein modeling (in 2022)</a>
  </div>
</li>
    </ul>
  </li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./links.html" class="sidebar-item-text sidebar-link active">Modeling Resources</a>
  </div>
</li>
    </ul>
    </div>
</nav>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#template-search" id="toc-template-search" class="nav-link active" data-scroll-target="#template-search"><strong>1. Template search</strong></a></li>
  <li><a href="#secondary-structure-and-1d-features-services" id="toc-secondary-structure-and-1d-features-services" class="nav-link" data-scroll-target="#secondary-structure-and-1d-features-services"><strong>2. Secondary structure and 1D features services</strong></a></li>
  <li><a href="#protein-modeling" id="toc-protein-modeling" class="nav-link" data-scroll-target="#protein-modeling"><strong>3. Protein modeling</strong></a></li>
  <li><a href="#model-quality-assessment" id="toc-model-quality-assessment" class="nav-link" data-scroll-target="#model-quality-assessment"><strong>4. Model quality assessment</strong></a></li>
  <li><a href="#other-resources" id="toc-other-resources" class="nav-link" data-scroll-target="#other-resources"><strong>5. Other resources</strong></a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title d-none d-lg-block">Modeling Resources</h1>
</div>





<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Modesto Redrejo Rodríguez </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">September 21, 2022</p>
    </div>
  </div>
    
  </div>
  

</header>

<div class="callout-note callout callout-style-default callout-captioned">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-caption-container flex-fill">
This is not a comprehensive list of Protein Modeling resources. Rather, it contains some of the references, methods, and online servers that I found useful through the years.
</div>
</div>
<div class="callout-body-container callout-body">
<p>Most useful links (IMHO) are highlighted in bold. Other (many) tools may be useful for you, feel free to let me know and I’ll update the list.</p>
</div>
</div>
<p>Links accessed on 21/September/2022</p>
<section id="template-search" class="level3">
<h3 class="anchored" data-anchor-id="template-search"><strong>1. Template search</strong></h3>
<ul>
<li><p><strong>BLAST</strong>: <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi" class="uri">https://blast.ncbi.nlm.nih.gov/Blast.cgi</a></p></li>
<li><p><strong>HHPRED</strong>: <a href="https://toolkit.tuebingen.mpg.de/tools/hhpred" class="uri">https://toolkit.tuebingen.mpg.de/tools/hhpred</a></p></li>
<li><p><strong>JackHMMER</strong>: <a href="https://www.ebi.ac.uk/Tools/hmmer/search/jackhmmer" class="uri">https://www.ebi.ac.uk/Tools/hmmer/search/jackhmmer</a></p>
<ul>
<li>Ref. <a href="https://nar.oxfordjournals.org/content/46/W1/W200" class="uri">https://nar.oxfordjournals.org/content/46/W1/W200</a></li>
</ul></li>
</ul>
</section>
<section id="secondary-structure-and-1d-features-services" class="level3">
<h3 class="anchored" data-anchor-id="secondary-structure-and-1d-features-services"><strong>2. Secondary structure and 1D features services</strong></h3>
<ul>
<li><p><strong>Jpred4</strong>: <a href="http://www.compbio.dundee.ac.uk/jpred/" class="uri">http://www.compbio.dundee.ac.uk/jpred/</a></p>
<ul>
<li><p>Secondary structure prediction</p></li>
<li><p>Ref. <a href="http://nar.oxfordjournals.org/content/early/2015/04/16/nar.gkv332" class="uri">http://nar.oxfordjournals.org/content/early/2015/04/16/nar.gkv332</a></p></li>
</ul></li>
<li><p><strong>PsiPred</strong> : <a href="http://bioinf.cs.ucl.ac.uk/psipred/" class="uri">http://bioinf.cs.ucl.ac.uk/psipred/</a></p>
<ul>
<li><p>Secondary structure prediction</p></li>
<li><p>MEMSAT-SVM (Membrane Helix Prediction)</p></li>
<li><p>Ref: <a href="https://academic.oup.com/nar/article/47/W1/W402/5480136" class="uri">https://academic.oup.com/nar/article/47/W1/W402/5480136</a></p></li>
</ul></li>
<li><p><strong>MULTICOM Toolbox</strong>: <a href="http://sysbio.rnet.missouri.edu/multicom_toolbox/web%20services.html" class="uri">http://sysbio.rnet.missouri.edu/multicom_toolbox/web%20services.html</a></p>
<ul>
<li><p>&nbsp;Pulls together protein structure and structural feature prediction tools.</p></li>
<li><p>Server currently offline. Application available on GitHub: <a href="https://github.com/multicom-toolbox" class="uri">https://github.com/multicom-toolbox</a></p></li>
</ul></li>
<li><p><strong>SCRATCH</strong>: <a href="http://scratch.proteomics.ics.uci.edu/" class="uri">http://scratch.proteomics.ics.uci.edu/</a></p>
<ul>
<li>Metaserver with a wide range of 1D-3D prediction tools.</li>
</ul></li>
<li><p><strong>SYMPRED</strong>: <a href="http://www.ibi.vu.nl/programs/sympredwww/" class="uri">http://www.ibi.vu.nl/programs/sympredwww/</a></p>
<ul>
<li>Consensus secondary structure prediction</li>
</ul></li>
<li><p>Genesilico: <a href="http://genesilico.pl/" class="uri">http://genesilico.pl/</a></p>
<ul>
<li>Secondary structure prediction metaserver (currently out of service!)</li>
</ul></li>
<li><p><strong>MESSA</strong>: <a href="http://prodata.swmed.edu/MESSA/MESSA.cgi" class="uri">http://prodata.swmed.edu/MESSA/MESSA.cgi</a></p>
<ul>
<li><p>1D features Metaserver</p></li>
<li><p>Ref.<a href="https://bmcbiol.biomedcentral.com/articles/10.1186/1741-7007-10-82" class="uri">https://bmcbiol.biomedcentral.com/articles/10.1186/1741-7007-10-82</a></p></li>
</ul></li>
<li><p><strong>DeepTMHMM</strong>: <a href="https://dtu.biolib.com/DeepTMHMM" class="uri">https://dtu.biolib.com/DeepTMHMM</a></p>
<ul>
<li><p>Deep-learning 2022 version of the classic TMHMM predictor of transmembrane helices</p></li>
<li><p>Ref. <a href="https://www.biorxiv.org/content/10.1101/2022.04.08.487609v1" class="uri">https://www.biorxiv.org/content/10.1101/2022.04.08.487609v1</a></p></li>
</ul></li>
<li><p><strong>Phobius</strong>: <a href="https://www.ebi.ac.uk/Tools/pfa/phobius/" class="uri">https://www.ebi.ac.uk/Tools/pfa/phobius/</a></p>
<ul>
<li>Prediction of transmembrane topology and signal peptides from the amino acid sequence of a protein.</li>
</ul></li>
<li><p><strong>TopCons</strong>: <a href="http://topcons.cbr.su.se/" class="uri">http://topcons.cbr.su.se/</a></p>
<ul>
<li><p>Consensus prediction of membrane protein topology and signal peptides</p></li>
<li><p>Ref. <a href="https://academic.oup.com/nar/article-lookup/doi/10.1093/nar/gkv485" class="uri">https://academic.oup.com/nar/article-lookup/doi/10.1093/nar/gkv485</a></p></li>
</ul></li>
<li><p><strong>PSORT</strong>: <a href="https://psort.hgc.jp/" class="uri">https://psort.hgc.jp/</a></p>
<ul>
<li>Several localization signals</li>
</ul></li>
<li><p><strong>PRED-TMMB:</strong> <a href="http://bioinformatics.biol.uoa.gr/PRED-TMBB/" class="uri">http://bioinformatics.biol.uoa.gr/PRED-TMBB/</a></p>
<ul>
<li>Predicion of beta-barrel TMEs &amp; topology (HMM-based)</li>
</ul></li>
<li><p><strong>PROF-TMF</strong>: <a href="https://open.predictprotein.org/" class="uri">https://open.predictprotein.org/</a></p>
<ul>
<li>Predicion of beta-barrel TMEs (PredictProtein Suite)</li>
</ul></li>
<li><p><strong>NLStradamus</strong>: <a href="http://www.moseslab.csb.utoronto.ca/NLStradamus/" class="uri">http://www.moseslab.csb.utoronto.ca/NLStradamus/</a></p>
<ul>
<li>Prediction of NLS</li>
</ul></li>
<li><p><strong>Sulfinator:</strong> <a href="https://web.expasy.org/sulfinator/" class="uri">https://web.expasy.org/sulfinator/</a></p>
<ul>
<li>Prediction of disulfide bonds</li>
</ul></li>
<li><p><strong>DTU Tech resources</strong>: <a href="https://services.healthtech.dtu.dk/" class="uri">https://services.healthtech.dtu.dk/</a></p>
<ul>
<li>A collection of molecular biology tools, includin a number of services for 1D features prediction<br>
</li>
</ul></li>
</ul>
</section>
<section id="protein-modeling" class="level3">
<h3 class="anchored" data-anchor-id="protein-modeling"><strong>3. Protein modeling</strong></h3>
<ul>
<li><p><strong>SwissModel</strong>: <a href="https://swissmodel.expasy.org/" class="uri">https://swissmodel.expasy.org/</a></p>
<ul>
<li>Ref: <a href="https://academic.oup.com/nar/article-lookup/doi/10.1093/nar/gky427" class="uri">https://academic.oup.com/nar/article-lookup/doi/10.1093/nar/gky427</a></li>
</ul></li>
<li><p><strong>Robetta</strong> server: <a href="https://robetta.bakerlab.org/" class="uri">https://robetta.bakerlab.org/</a></p>
<ul>
<li>Several methods</li>
</ul></li>
<li><p><strong>ModWeb:</strong> <a href="https://modbase.compbio.ucsf.edu/modweb/" class="uri">https://modbase.compbio.ucsf.edu/modweb/</a></p>
<ul>
<li><p>Webserver using MODELLER (free registration required)</p></li>
<li><p>Ref. <a href="https://pubmed.ncbi.nlm.nih.gov/24271400/" class="uri">https://pubmed.ncbi.nlm.nih.gov/24271400/</a></p></li>
</ul></li>
<li><p><strong>ColabFold</strong>: <a href="https://github.com/sokrypton/ColabFold" class="uri">https://github.com/sokrypton/ColabFold</a></p>
<ul>
<li><p>Several notebooks available</p></li>
<li><p>Ref. <a href="https://www.nature.com/articles/s41592-022-01488-1" class="uri">https://www.nature.com/articles/s41592-022-01488-1</a></p></li>
</ul></li>
<li><p>CPHmodels: <a href="https://services.healthtech.dtu.dk/service.php?CPHmodels-3.2" class="uri">https://services.healthtech.dtu.dk/service.php?CPHmodels-3.2</a></p>
<ul>
<li>Ref: <a href="https://academic.oup.com/nar/article-lookup/doi/10.1093/nar/gkq535" class="uri">https://academic.oup.com/nar/article-lookup/doi/10.1093/nar/gkq535</a></li>
</ul></li>
<li><p><strong>Zhang server</strong>: <a href="https://zhanggroup.org/services/" class="uri">https://zhanggroup.org/services/</a></p>
<ul>
<li><p>Multiple methods, including I-TASSER, C-I-TASSER or I-TASSER-MTD</p></li>
<li><p>See references within.</p></li>
</ul></li>
<li><p>PsiPRED: <a href="http://bioinf.cs.ucl.ac.uk/psipred/" class="uri">http://bioinf.cs.ucl.ac.uk/psipred/</a></p>
<ul>
<li><p>Fold recognition server (among other tools)</p></li>
<li><p>Ref: <a href="https://academic.oup.com/nar/article/47/W1/W402/5480136" class="uri">https://academic.oup.com/nar/article/47/W1/W402/5480136</a></p></li>
</ul></li>
<li><p>Multicom: <a href="http://sysbio.rnet.missouri.edu/multicom_cluster/" class="uri">http://sysbio.rnet.missouri.edu/multicom_cluster/</a></p>
<ul>
<li>Ref.<a href="https://bmcstructbiol.biomedcentral.com/articles/10.1186/1472-6807-13-2" class="uri">https://bmcstructbiol.biomedcentral.com/articles/10.1186/1472-6807-13-2</a></li>
</ul></li>
<li><p><strong>RaptorX</strong>: <a href="http://raptorx.uchicago.edu/ContactMap/" class="uri">http://raptorx.uchicago.edu/ContactMap/</a></p>
<ul>
<li><p>Distance-based Protein Folding Powered by Deep Learning</p></li>
<li><p>Ref. <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1005324" class="uri">https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1005324</a></p></li>
</ul></li>
<li><p><strong>Phyre2</strong>: <a href="http://www.sbg.bio.ic.ac.uk/~phyre2/html/page.cgi?id=index" class="uri">http://www.sbg.bio.ic.ac.uk/~phyre2/html/page.cgi?id=index</a></p>
<ul>
<li><p>Ref. <a href="http://www.nature.com/nprot/journal/v10/n6/ris/nprot.2015.053.ris" class="uri">http://www.nature.com/nprot/journal/v10/n6/ris/nprot.2015.053.ris</a></p></li>
<li><p>Fold recognition &amp; fragment assembler<br>
</p></li>
</ul></li>
</ul>
</section>
<section id="model-quality-assessment" class="level3">
<h3 class="anchored" data-anchor-id="model-quality-assessment"><strong>4. Model quality assessment</strong></h3>
<ul>
<li><p><strong>QMEAN</strong>: <a href="https://swissmodel.expasy.org/qmean/" class="uri">https://swissmodel.expasy.org/qmean/</a></p>
<ul>
<li><p>Includes the new version QMEANDisCo</p></li>
<li><p>Ref. <a href="https://doi.org/10.1093/bioinformatics/btz828" class="uri">https://doi.org/10.1093/bioinformatics/btz828</a></p></li>
</ul></li>
<li><p><strong>Modfold: <a href="https://www.reading.ac.uk/bioinf/ModFOLD/ModFOLD6_form.html" class="uri">https://www.reading.ac.uk/bioinf/ModFOLD/ModFOLD6_form.html</a></strong></p>
<ul>
<li>Ref: <a href="https://academic.oup.com/nar/article-lookup/doi/10.1093/nar/gkx332" class="uri">https://academic.oup.com/nar/article-lookup/doi/10.1093/nar/gkx332</a></li>
</ul></li>
<li><p><strong>VoroMQA: <a href="http://bioinformatics.ibt.lt/wtsam/voromqa" class="uri">http://bioinformatics.ibt.lt/wtsam/voromqa</a></strong></p>
<ul>
<li>Ref: <a href="https://onlinelibrary.wiley.com/doi/full/10.1002/prot.25278" class="uri">https://onlinelibrary.wiley.com/doi/full/10.1002/prot.25278</a></li>
</ul></li>
<li><p>Saves: <a href="https://servicesn.mbi.ucla.edu/SAVES/" class="uri">https://servicesn.mbi.ucla.edu/SAVES/</a></p>
<ul>
<li>Quality assessment in several “classic” servers</li>
</ul></li>
<li><p>SIFT: <a href="http://sift.cchmc.org/" class="uri">http://sift.cchmc.org/</a></p>
<ul>
<li><p><strong>S</strong>equence <strong>I</strong>ndependent (as well as dependent) <strong>F</strong>iltering <strong>T</strong>ools for efficient and accurate protein model quality assessment</p></li>
<li><p>Ref. <a href="https://www.tandfonline.com/doi/abs/10.1080/00268970410001728636?journalCode=tmph20" class="uri">https://www.tandfonline.com/doi/abs/10.1080/00268970410001728636?journalCode=tmph20</a></p></li>
</ul></li>
</ul>
</section>
<section id="other-resources" class="level3">
<h3 class="anchored" data-anchor-id="other-resources"><strong>5. Other resources</strong></h3>
<ul>
<li><p><strong>DALI</strong>: <a href="http://ekhidna2.biocenter.helsinki.fi/dali/" class="uri">http://ekhidna2.biocenter.helsinki.fi/dali/</a></p>
<ul>
<li><p>Web service for comparing protein structures in 3D</p></li>
<li><p>Ref:<a href="https://onlinelibrary.wiley.com/doi/full/10.1002/pro.3749" class="uri">https://onlinelibrary.wiley.com/doi/full/10.1002/pro.3749</a></p></li>
</ul></li>
<li><p><strong>FoldSeek</strong>: <a href="https://search.foldseek.com/search" class="uri">https://search.foldseek.com/search</a></p>
<ul>
<li><p>Find structural homologs</p></li>
<li><p>Ref. <a href="https://www.biorxiv.org/content/10.1101/2022.02.07.479398v4" class="uri">https://www.biorxiv.org/content/10.1101/2022.02.07.479398v4</a></p></li>
</ul></li>
<li><p>FatCat<strong>:</strong> <a href="http://fatcat.godziklab.org/fatcat/fatcat_pair.html" class="uri">http://fatcat.godziklab.org/fatcat/fatcat_pair.html</a></p>
<ul>
<li><p>Pairwise structural alignment of PDB entries or .pdb files.</p></li>
<li><p>Ref. <a href="https://academic.oup.com/nar/article-lookup/doi/10.1093/nar/gkaa443" class="uri">https://academic.oup.com/nar/article-lookup/doi/10.1093/nar/gkaa443</a></p></li>
</ul></li>
<li><p>LGA: <a href="http://proteinmodel.org/AS2TS/LGA/lga.html" class="uri">http://proteinmodel.org/AS2TS/LGA/lga.html</a></p>
<ul>
<li><p>LGA&nbsp;program is being developed for&nbsp;structure&nbsp;comparative analysis of two selected 3D&nbsp;protein&nbsp;structures or fragments of 3D&nbsp;protein&nbsp;structures</p></li>
<li><p>Ref. <a href="https://academic.oup.com/nar/article-lookup/doi/10.1093/nar/gkg571" class="uri">https://academic.oup.com/nar/article-lookup/doi/10.1093/nar/gkg571</a></p></li>
</ul></li>
<li><p>Galaxy Refine: <a href="http://galaxy.seoklab.org/cgi-bin/submit.cgi?type=REFINE" class="uri">http://galaxy.seoklab.org/cgi-bin/submit.cgi?type=REFINE</a></p>
<ul>
<li><p>A server for automatically refining protein model structures.</p></li>
<li><p>Ref. <a href="http://www.ncbi.nlm.nih.gov/pubmed/23737448" class="uri">http://www.ncbi.nlm.nih.gov/pubmed/23737448</a></p></li>
</ul></li>
<li><p>Galaxy Refine2: <a href="http://galaxy.seoklab.org/cgi-bin/submit.cgi?type=REFINE2"><u>galaxy.seoklab.org/cgi-bin/submit.cgi?type=REFINE2</u></a></p>
<ul>
<li><p>&lt;300 aa</p></li>
<li><p>Ref:<a href="https://academic.oup.com/nar/article/47/W1/W451/5475172" class="uri">https://academic.oup.com/nar/article/47/W1/W451/5475172</a></p></li>
</ul></li>
<li><p>HOMCOS: <a href="http://homcos.pdbj.org/" class="uri">http://homcos.pdbj.org/</a></p>
<ul>
<li><p>Modeling of 3D Structures of Complexes</p></li>
<li><p>Ref. <a href="http://link.springer.com/article/10.1007/s10969-016-9208-y?wt_mc=Internal.Event.1.SEM.ArticleAuthorOnlineFirst" class="uri">http://link.springer.com/article/10.1007/s10969-016-9208-y?wt_mc=Internal.Event.1.SEM.ArticleAuthorOnlineFirst</a></p></li>
</ul></li>
<li><p>pyDockWEB: <a href="https://life.bsc.es/pid/pydockweb" class="uri">https://life.bsc.es/pid/pydockweb</a></p>
<ul>
<li>Prediction of protein-protein interactions (hetero-oligomers modeling)</li>
</ul></li>
<li><p>Hex: <a href="http://hexserver.loria.fr/" class="uri">http://hexserver.loria.fr/</a></p>
<ul>
<li><p>Prediction of protein-protein interactions (homo- &amp; hetero-oligomers modeling)</p></li>
<li><p>Ref. <a href="https://onlinelibrary.wiley.com/doi/full/10.1002/prot.24433" class="uri">https://onlinelibrary.wiley.com/doi/full/10.1002/prot.24433</a></p></li>
</ul></li>
<li><p>DaReUs-Loop<strong>:</strong> <a href="https://bioserv.rpbs.univ-paris-diderot.fr/services/DaReUS-Loop/" class="uri">https://bioserv.rpbs.univ-paris-diderot.fr/services/DaReUS-Loop/</a></p>
<ul>
<li><p>Modeling loops</p></li>
<li><p>Ref. <a href="https://www.nature.com/articles/s41598-018-32079-w" class="uri">https://www.nature.com/articles/s41598-018-32079-w</a></p></li>
</ul></li>
<li><p>GalaxyLoop: <a href="http://galaxy.seoklab.org/cgi-bin/submit.cgi?type=LOOP" class="uri">http://galaxy.seoklab.org/cgi-bin/submit.cgi?type=LOOP</a></p>
<ul>
<li><p>Loops refining</p></li>
<li><p>Ref. <a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0113811" class="uri">https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0113811</a></p></li>
</ul></li>
<li><p>Expasy proteomics tools: <a href="https://www.expasy.org/proteomics" class="uri">https://www.expasy.org/proteomics</a></p>
<ul>
<li>List of qMultiple links and resources</li>
</ul></li>
<li><p>Protein modeling links @Sali Lab: <a href="https://salilab.org/bioinformatics_resources.html" class="uri">https://salilab.org/bioinformatics_resources.html</a></p></li>
<li><p>Homology modeling @Proteopedia:</p>
<ul>
<li><p><a href="https://proteopedia.org/wiki/index.php/Homology_model" class="uri">https://proteopedia.org/wiki/index.php/Homology_model</a></p></li>
<li><p><a href="https://proteopedia.org/wiki/index.php/User:Wayne_Decatur/Homology_Modeling" class="uri">https://proteopedia.org/wiki/index.php/User:Wayne_Decatur/Homology_Modeling</a></p></li>
</ul></li>
<li><p>List of protein structure prediction software @Wikipedia: <a href="https://en.wikipedia.org/wiki/List_of_protein_structure_prediction_software" class="uri">https://en.wikipedia.org/wiki/List_of_protein_structure_prediction_software</a></p></li>
</ul>


</section>

</main> <!-- /main -->
<script type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      let href = ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
    var links = window.document.querySelectorAll('a:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
          // target, if specified
          link.setAttribute("target", "_blank");
          // default icon
          link.classList.add("external");
      }
    }
});
</script>
</div> <!-- /content -->



</body></html>